test/org/sagebionetworks/bridge/services/HealthDataServiceTest.java

```diff
@@ -123,4 +140,43 @@ public void getRecordsForUploadDateSuccess() {
         assertEquals("bar healthcode", recordList.get(1).getHealthCode());
         assertEquals("baz healthcode", recordList.get(2).getHealthCode());
     }
+
+    @Test
+    public void updateRecordSuccess() throws Exception {
+
+        // first create a mock record
+        // record
+        HealthDataRecord record = DAO.getRecordBuilder().withHealthCode(TEST_HEALTH_CODE)
+                .withSchemaId(TEST_SCHEMA_ID).withSchemaRevision(3).withStudyId(TEST_STUDY_ID).build();
+
+        // mock dao
+        HealthDataDao mockDao = mock(HealthDataDao.class);
+        when(mockDao.createOrUpdateRecord(record)).thenReturn(TEST_RECORD_ID);
+
+        HealthDataService svc = new HealthDataService();
+        svc.setHealthDataDao(mockDao);
+
+        // execute and validate
+        String retVal = svc.createOrUpdateRecord(record);
+        assertEquals(TEST_RECORD_ID, retVal);
+
+        // then create a mock json request
+        RecordExportStatusRequest recordExportStatusRequest = createMockRecordExportStatusRequest();
+        when(mockDao.getRecordById(TEST_RECORD_ID)).thenReturn(record);
+
+        // finally call service method and assert
+        svc.updateRecordsWithExporterStatus(recordExportStatusRequest);
+        HealthDataRecord recordAfter = svc.getRecordById(TEST_RECORD_ID);
+        assertNotNull(recordAfter.getSynapseExporterStatus());
+        assertEquals(HealthDataRecord.ExporterStatus.SUCCEEDED, recordAfter.getSynapseExporterStatus());
```

Comment:
This is a little more complicated. Yes, it's important to test that the Service is correctly setting the ExportStatus on the record. However, it looks like the implementation just sets the status on the record and throws it away without saving it to DDB.

It also demonstrates the importance of manual tests. (Or integration tests if you're modifying an existing API, but that comes later.) Your unit tests and mocks make certain assumptions about how a system works. However, the real system might work under different assumptions. As such, it's important to test things manually, at a high level, just to make sure everything works they way you expected them to.